dts/bindings/gpio/nordic,npm1300-gpio.yaml

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: nPM1300 GPIO Controller
+
+compatible: "nordic,npm1300-gpio"
```

Comment:
After checking the datasheet, I wonder if we should split the gpio0..4 pins from led0..2 into 2 GPIO controllers. I'd expect, for example, to configure in DT stuff like functions of the LEDs, something like:

```
led-controller {
    compatible = "nordic,npm1300-led";
    gpio-controller;
    ngpios = <3>;
    nordic,led0-mode = <NPM1300_LED_MODE_ERROR>;
    ...
};
```

then driver init call would setup registers accordingly, and forbid usage of such pins at runtime. I suppose it is not strictly necessary to split into 2 controllers, though. Just that I observe LED/GPIO IP blocks are kind of independent.

Reply:
note: similar comment for the GPIO block, we could add properties to configure some modes. It can be added in a 2nd round, though.

Reply:
Nice idea.  The combining is a bit clunky, and they have slightly different behaviour so it would be clearer to split them.

Reply:
Is it better to do this as part of the PR, or remove LEDs and add them as a separate PR?

Reply:
It's ok to add it here, if it grows too much it can be split later.

Reply:
I'm going to split this out into a later PR.  For this initial release it's not essential, as the LEDs work automatically by default.